controllers/common.go

```diff
@@ -13,12 +34,404 @@ func getServiceAccountName() string {
+		}
+		return os.Getenv("INDEXER_IMAGE")
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.ImageOverride != "" {
+			return instance.Spec.Deployments.Database.ImageOverride
+		}
+		return os.Getenv("POSTGRES_IMAGE")
+	}
+	log.V(2).Info("Unknown deployment %s ", deploymentName)
+	return ""
+}
+
+func hasDeployments(instance *searchv1alpha1.Search) bool {
+	return instance.Spec.Deployments.DeepCopy() != nil
+
+}
+
+func isDeploymentCustomized(deploymentName string, instance *searchv1alpha1.Search) bool {
+	if !hasDeployments(instance) {
+		return false
+	}
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.DeepCopy() != nil {
+			return true
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.DeepCopy() != nil {
+			return true
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.DeepCopy() != nil {
+			return true
+		}
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.DeepCopy() != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func isResourcesCustomized(deploymentName string, instance *searchv1alpha1.Search) bool {
+	if !isDeploymentCustomized(deploymentName, instance) {
+		return false
+	}
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.Resources != nil {
+			return true
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.Resources != nil {
+			return true
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.Resources != nil {
+			return true
+		}
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.Resources != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *SearchReconciler) createOrUpdateConfigMap(ctx context.Context, cm *corev1.ConfigMap) (*reconcile.Result, error) {
+	found := &corev1.ConfigMap{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Name:      cm.Name,
+		Namespace: cm.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(context.TODO(), cm)
+		if err != nil {
+			log.Error(err, "Could not create configmap")
+			return &reconcile.Result{}, err
+		}
+	}
+	log.V(2).Info("Created %s configmap", cm.Name)
+	return nil, nil
+}
+
+func (r *SearchReconciler) createOrUpdateDeployment(ctx context.Context, deploy *appsv1.Deployment) (*reconcile.Result, error) {
+	found := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Name:      deploy.Name,
+		Namespace: deploy.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(context.TODO(), deploy)
+		if err != nil {
+			log.Error(err, "Could not create deployment")
+			return &reconcile.Result{}, err
+		}
+	}
+	if err := r.Update(context.TODO(), deploy); err != nil {
+		log.Error(err, "Could not update deployment")
+		return &reconcile.Result{}, err
+	}
+	log.V(2).Info("Created %s deployment", deploy.Name)
```

Comment:
This log could be inaccurate when the deployment is updated.

Reply:
Would it help to also log the entire object **when verbosity is high?**
```go
// We could add both logs at different verbosity levels.
log.V(2).Info("Created %s deployment", deploy.Name)
log.V(9).Info("Created deployment. %+v, deploy)
```

Reply:
Fixed it